### PR TITLE
Fix flakey SpanTest by allow better validation on FakeSpanExporter

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Track the count of the numbers of spans exported and a method that waits for that value or times out. This fixes the flakiness for SpanTest but TracingApiTest is still flakey, which actually might be pointing to an actual race condition in the code that still needs to be addressed.

## Testing

Existing tests shouldn't be flakey

### CODE REVIEW

**Comment by bidetofevil** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/401?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #391** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/391?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #392** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/392?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #393** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/393?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #394** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/394?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #395** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/395?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #396** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/396?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #397** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/397?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #401** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/401?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                  * **PR #402** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/402?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/401?utm_source=stack-comment).

File: embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/SpanTest.kt [hunk 1]
```diff
@@ -47,17 +49,45 @@ internal class SpanTest {
...
+        }
         return CompletableResultCode.ofSuccess()
     }
 
     override fun shutdown(): CompletableResultCode {
         return CompletableResultCode.ofSuccess()
     }
+
+    /**
+     * Block the thread until the number of spans expected have been exported by this exporter since the last flush
```
Comment (bidetofevil) [inline embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/SpanTest.kt:79]: Theres's a small bug in this given how this function's behaviour is defined: when the exporter is flushed, the latch countdown values are persisted, but the comparison is for the number of exported spans AFTER the flush. 

So basically the latches take into account flushed spans, while invocations that happen after won't.

This doesn't affect this test, but the inconsistency is bad, so I'll be fixing this in the next PR to always look at all spans exported instead even before the flush.

Comment (bidetofevil) [reply embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/SpanTest.kt]: Never mind - I'm merging the two fixes into one PR

**Review by fractalwrench — APPROVED** [review]
LGTM

**Comment by bidetofevil** [comment]
### Merge activity

* **Feb 14, 2:46 PM EST**: @bidetofevil started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/401).

**Pull request merged**